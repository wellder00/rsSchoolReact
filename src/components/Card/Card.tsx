import React from 'react';
// import Link  from 'next/link';
import Image from 'next/image';
import notFound from '../../../public/notFound.png';
import { Pokemon } from '../../types/interfaces';

// import { useAppSelector } from '@/lib/redux/hooks/reduxHooks';
import styles from './Card.module.scss';

type Props = {
  data: Pokemon | null | undefined;
  pathname?: string;
};

const Card: React.FC<Props> = ({ data, pathname }) => {
  // const currentPage = useAppSelector((state) => state.itemsAmount.currentPage);

  // const queryParams = `about_character/${data?.id}/?page=${currentPage}`;
  return (
    // <Link className={styles.link} to={queryParams}>
    <div className={`${pathname === '/' ? styles.card : styles.miniCard}`}>
      <div className={styles.characterInfo}>
        <h2 className={`${pathname === '/' ? styles.title : styles.miniTitle}`}>{data?.name}</h2>
        <div className={styles.characteristic}>Weight: {data?.weight}</div>
        <div className={styles.characteristic}>Species: {data?.species}</div>
      </div>
      <Image className={styles.characterImage} src={data?.sprites || notFound} alt={'data?.name'} />
    </div>
    // </Link>
  );
};

export default Card;
