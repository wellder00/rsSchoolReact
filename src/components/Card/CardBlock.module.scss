.wrapper {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  gap: 20px;
}

.wrapCard {
  padding: 10px;
  border: 1px solid green;
  transition: transform 0.3s ease;

  &:hover {
    transform: scale(1.05);
  }
}

.loadingIcon { 
  font-size: 25px;
}

