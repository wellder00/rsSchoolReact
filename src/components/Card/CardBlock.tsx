import React, { useState, useEffect } from 'react';

import styles from './CardBlock.module.scss';

import { Loader } from '../Loader';
import { NotFound } from '../NotFound';

import { Character, Info } from '../../types/interfaces';

type Props = {
  rickAndMortyData: Info<Character> | null;
};

const CardBlock: React.FC<Props> = ({ rickAndMortyData }) => {
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    setLoading(false);
  }, [rickAndMortyData]);

  const renderCharacterCard = (data: Character) => (
    <div className={styles.card} key={data.id}>
      <div className={styles.characterInfo}>
        <h2 className={styles.title}>{data.name}</h2>
        <div>Status: {data.status}</div>
        <div>Species: {data.species}</div>
        <div>Gender: {data.gender}</div>
      </div>
      <img className={styles.characterImage} src={data.image} alt={data.name} />
    </div>
  );

  if (loading) {
    return <Loader />;
  }

  if (!rickAndMortyData) {
    return <NotFound />;
  }

  return (
    <div className={styles.wrapper}>
      {rickAndMortyData.results?.map((data) => renderCharacterCard(data))}
    </div>
  );
};

export default CardBlock;
