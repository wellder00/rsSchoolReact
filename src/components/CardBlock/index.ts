export { default as CardBlock } from './CardBlock';
