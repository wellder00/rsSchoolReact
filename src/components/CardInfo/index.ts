export { default as CardInfo } from './CardInfo';
