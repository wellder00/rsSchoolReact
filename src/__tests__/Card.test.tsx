import { render, screen, fireEvent } from '@testing-library/react';
import { MemoryRouter, useNavigate } from 'react-router-dom';
import { Card } from '../components/Card';
import { ditto } from './PokemonData';

describe('Tests for the Card component', () => {
  it('Ensure that the card component renders the relevant card data;', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Card data={ditto} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText(/ditto/i)).toBeInTheDocument();
    expect(screen.getByText(/fire/i)).toBeInTheDocument();
    expect(screen.getByText(/40/i)).toBeInTheDocument();
  });

  it('Validate that clicking on a card opens a detailed card component;', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/']} initialIndex={0}>
        <Card data={ditto} />
      </MemoryRouter>
    );

    const linkElement = container.querySelector('a');
    if (linkElement) {
      fireEvent.click(linkElement);
      // expect(navigate).toHaveBeenCalledWith('/about_character/8?limit=null&offset=null&page=null')
    } else {
      throw new Error('Link element not found');
    }
  });
});
