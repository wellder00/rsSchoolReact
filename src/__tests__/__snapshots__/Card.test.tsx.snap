// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests for the Card component > Ensure that the card component renders the relevant card data; 1`] = `
<DocumentFragment>
  <a
    class="_link_d75732"
    href="/about_character/8?limit=null&offset=null&page=null"
  >
    <div
      class="_miniCard_d75732"
    >
      <div
        class="_characterInfo_d75732"
      >
        <h2
          class="_miniTitle_d75732"
        >
          ditto
        </h2>
        <div
          class="_characteristic_d75732"
        >
          Weight: 40
        </div>
        <div
          class="_characteristic_d75732"
        >
          Species: fire
        </div>
      </div>
      <img
        alt="ditto"
        class="_characterImage_d75732"
        src="/src/assets/images/teleportBottom.png"
      />
    </div>
  </a>
</DocumentFragment>
`;
